samples/demand_paging/README.rst

```diff
@@ -0,0 +1,49 @@
+.. _hello_world:
+
+Demand Paging
+#############
```

Comment:
this will help with discoverability by showing the sample there: https://docs.zephyrproject.org/latest/kernel/memory_management/demand_paging.html#api-reference

```suggestion
.. zephyr:code-sample:: demand-paging
   :name: Demand paging
   :relevant-api: mem-demand-paging

   Leverage demand paging to deal with code bigger than available RAM.
```